src/shogun/mathematics/Seedable.h

```diff
@@ -45,12 +45,19 @@ namespace shogun
 			return "Seedable";
 		}
 
+		bool is_set_seed() const
+		{
+			return m_is_seeded;
+		}
+
 	private:
 		void init()
 		{
 			Parent::watch_param(random::kSeed, &m_seed);
 			Parent::add_callback_function(
 			    random::kSeed, std::bind(seed_callback, this, std::ref(m_seed)));
+			Parent::add_callback_function(
+			    random::kSeed, [&]() { m_is_seeded = true; });
```

Comment:
So this is to avoid overriding the seed of the subobject?

Reply:
yes